kpops/utils/pydantic.py

```diff
@@ -111,20 +113,55 @@ def json_schema_extra(schema: dict[str, Any], model: type[BaseModel]) -> None:
 class YamlConfigSettingsSource(PydanticBaseSettingsSource):
     """Loads variables from a YAML file at the project's root."""
 
-    path_to_config = Path("config.yaml")
+    log = logging.getLogger()
+
+    config_dir = Path()
+    config_file_base_name = "config"
+    environment: str | None = None
+
+    def __init__(self, settings_cls) -> None:
+        super().__init__(settings_cls)
+        self.is_config_dir_valid()
+        default_config = self.load_config(
+            self.config_dir / f"{self.config_file_base_name}.yaml"
+        )
+        env_config = (
+            self.load_config(
+                self.config_dir
+                / f"{self.config_file_base_name}_{self.environment}.yaml"
+            )
+            if self.environment
+            else {}
+        )
+        self.config = update_nested_pair(env_config, default_config)
+
+    def is_config_dir_valid(self) -> None:
```

Comment:
methods named `is_<some-condition>` usually indicate a boolean return value. Since we don't return anything here we should consider naming it differently, e.g. `verify_config_dir_valid`

Reply:
How about `check_config_dir_exists`?